CONTRIBUTING.md

```diff
@@ -4,8 +4,50 @@ See [Superchain Upgrades] OP-Stack specifications.
 
 [Superchain Upgrades]: https://specs.optimism.io/protocol/superchain-upgrades.html
 
+
+## Adding a chain
+
+
+### 1. Set env vars
+
+To contribute a full OP-Stack chain configuration, the following data is required: contracts deployment, rollup config, L2 genesis. We provide a tool to scrape this information from your local monorepo folder.
+
+First, make a copy of `.env.example` named `.env`, and alter the variables to appropriate values.
+
+### 2. Run script
+Then, run
+
+```shell
+sh scripts/add-chain.sh
+```
+
+### 3. Understand output
+The tool will write the following data:
+- The main configuration source, with genesis data, and address of onchain system configuration.
+- Addresses of L1 contracts. (Note that all L2 addresses are statically known addresses defined in the OP-Stack specification, and thus not configured per chain.)
+- Genesis system config data
+- Compressed `genesis.json` definitions (in the `extra/genesis` directory) which pull in the bytecode by hash
+
+The genesis largely consists of contracts common with other chains:
+all contract bytecode is deduplicated and hosted in the `extra/bytecodes` directory.
+
+The format is a gzipped JSON `genesis.json` file, with either:
+- a `alloc` attribute, structured like a standard `genesis.json`,
+  but with `codeHash` (bytes32, `keccak256` hash of contract code) attribute per account,
+  instead of the `code` attribute seen in standard Ethereum genesis definitions.
+- a `stateHash` attribute: to omit a large state (e.g. for networks with a re-genesis or migration history).
+  Nodes can load the genesis block header, and state-sync to complete the node initialization.
+
+### 4. Raise your Pull Request
+  Automated checks will run, and your PR will be reviewed in due course.
+
+
+
 ## Adding a superchain target
 
+> **Note**
+> This is an infrequent operation and unecessary if you are just looking to add a chain to an existing superchain.
```

Comment:
```suggestion
> This is an infrequent operation and unnecessary if you are just looking to add a new L2 chain.
```

Reply:
One of the scenarios we want to cover is the following: 

* say we already have a `holesky` superchain in place
* we launch a `holesky-dev-0/op-labs-dev-0` devnet chain and want to add that to the registry
* someone needs to follow the steps in this section despite "just looking to add a new L2 chain to an existing L1 chain"
* in short, superchain != L1

Reply:
Sorry, maybe this is my lack of familiarity with devnets, but why isn't adding `holesky-dev-0` effectively adding a new devnet L1?

Reply:
ahh yes that does make sense, thank you!